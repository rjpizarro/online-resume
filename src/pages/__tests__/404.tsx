import React from "react"
import renderer from "react-test-renderer"
import { render, waitFor } from "@testing-library/react"
import * as Gatsby from "gatsby"


import NotFoundPage from "../404"

jest.mock("../../components/layout", () => (props) => <div>{props.children}</div>)

const data = {
  "site": {
    "siteMetadata": {
      "title": "Rodrigo Pizarro - Résumé",
      "description": "",
      "author": "@rjpizarro"
    }
  }
}

beforeEach(() => {
  Gatsby.useStaticQuery.mockImplementation(() => (data));
})

afterEach(() => {
  Gatsby.useStaticQuery.mockImplementation(() => {});
})

describe("404 Page", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<NotFoundPage />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("should configure page title correctly", async () => {
    render(<NotFoundPage />)

    await waitFor(() => {
      expect(document.title).toEqual("404: Not found | Rodrigo Pizarro - Résumé")
    })
  })
})