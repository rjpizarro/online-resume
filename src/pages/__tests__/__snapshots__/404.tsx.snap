// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page renders correctly 1`] = `
<div>
  <h1>
    NOT FOUND
  </h1>
  <p>
    You just hit a route that doesn't exist... the sadness.
  </p>
</div>
`;
